src/Storages/ObjectStorage/DataLakes/IStorageDataLake.h

```diff
@@ -135,6 +164,23 @@ class IStorageDataLake final : public StorageObjectStorage
         }
     }
 
+    void updateExternalDynamicMetadata(ContextPtr context) override
+    {
+        if constexpr (std::is_same_v<DataLakeMetadata, IcebergMetadata>)
+        {
+            current_metadata = DataLakeMetadata::create(Storage::object_storage, base_configuration, context);
+            ColumnsDescription column_description{current_metadata->getTableSchema()};
+            StorageInMemoryMetadata metadata;
+            metadata.setColumns(std::move(column_description));
+            setInMemoryMetadata(metadata);
+            return;
+        }
+        throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Method updateExternalDynamicMetadata is not supported by storage {}", getName());
+    }
+
+    bool hasExternalDynamicMetadata() const override { return std::is_same_v<DataLakeMetadata, IcebergMetadata>; }
```

Comment:
Use virtual methods in `IDataLakeMetadata` instead of `is_same_v`

Reply:
Not relevant anymore